Add some details on "javascript" page

GeoSot opened the pull request:
This PR tries to help newcomers, adding some information on "getting-started => javascript"

Changes: 
* minor typos
* group jQuery stuff & add details of use
* add details about Data attributes usage (plus `data-bs-config`)
* explain the difference  `getInstance` & `getOrCreateInstance` methods
* add async usage example in addition with `dispose` method
* add shared methods & properties

Needs:
* [x] a native speaker to proofread the changes & additions

Preview : https://deploy-preview-36363--twbs-bootstrap.netlify.app/docs/5.2/getting-started/javascript/

GeoSot committed:
docs: Add more details on "getting-started -> javascript" page & group jQuery stuff

julien-deramond reviewed (commented):
Good idea to group and isolate jQuery stuff at the end! I find the content of this page clearer.

Sorry I've only seen that this PR was a draft at the end after having already created some comments. I still leave them here if they can be useful.

As you know I'm not a native speaker so take my feedback with caution :smile:

GeoSot commented on site/content/docs/5.2/getting-started/javascript.md:
```diff
@@ -47,18 +47,19 @@ Some plugins and CSS components depend on other plugins. If you include plugins
 
 Nearly all Bootstrap plugins can be enabled and configured through HTML alone with data attributes (our preferred way of using JavaScript functionality). Be sure to **only use one set of data attributes on a single element** (e.g., you cannot trigger a tooltip and modal from the same button.)
 
+Make sure to change the case type of the option name from camelCase to kebab-case when passing the options via data attributes. For example, use `data-bs-custom-class="beautifier"` instead of `data-bs-customClass="beautifier"`.
```
I want to use the partial too. The first sentence thought, seemed as repetition 

I'll take a look again

(THANKS for the feedback, it really helps 🤟 )

GeoSot committed:
Update after feedback

mdo committed:
Rewrite some content

- Shorten copy here and there
- Reduce some callouts that are entire sections with h2 headings
- Re-arrange a couple things

mdo commented:
Pushed a big copy update after walking through this. Appreciate the polish!

mdo committed:
Merge branch 'main' into gs/js-docs

mdo committed:
Merge branch 'main' into gs/js-docs

mdo committed:
Merge branch 'main' into gs/js-docs
